python/cuml/tsa/arima.pyx

```diff
@@ -406,25 +402,33 @@ class ARIMA(Base):
         criteria = [order.k, order.p, order.q, order.P, order.Q, True]
         for i in range(len(names)):
             if criteria[i] > 0:
-                params[names[i]] = getattr(self, names[i])
+                params[names[i]] = getattr(self, "{}_".format(names[i]))
         return params
 
-    def set_params(self, params: Mapping[str, object]):
-        """Set the parameters of the model
+    def set_fit_params(self, params: Mapping[str, object]):
+        """Set all the fit parameters. Not to be confused with set_params
+        Note: unpack() can be used to load a compact vector of the parameters
 
         Parameters
         ----------
-        params: Mapping[str, np.ndarray]
-            A mapping (e.g dictionary) of parameter names and associated arrays
+        params:
+            A dictionary of parameter names and associated arrays
             The key names are in {"mu", "ar", "ma", "sar", "sma", "sigma2"}
-            The shape of the arrays are (batch_size,) for mu parameters and
+            The shape of the arrays are (batch_size,) for mu and sigma2 and
             (n, batch_size) for any other type, where n is the corresponding
             number of parameters of this type.
         """
         for param_name in ["mu", "ar", "ma", "sar", "sma", "sigma2"]:
             if param_name in params:
-                array, _, _, _, _ = input_to_host_array(params[param_name])
-                setattr(self, param_name, array)
+                array, *_ = input_to_cuml_array(params[param_name],
+                                                check_dtype=np.float64)
+                setattr(self, "_{}_".format(param_name), array)
+
+    def get_params():
+        raise NotImplementedError
+
+    def set_params():
+        raise NotImplementedError
```

Comment:
It is better to instead overwrite `get_param_names()` and raise an exception there.